tools/github-actions/audit/src/reports.ts

```diff
@@ -0,0 +1,198 @@
+
+/**
+ * Severities supported by yarn npm audit from the lowest to the highest criticality
+ */
+export const severities = ['info', 'low', 'moderate', 'high', 'critical'] as const satisfies Severity[];
+
+/**
+ * Description of an advisory as it will be displayed in the action summary
+ */
+export interface OtterAdvisory {
+  severity: Severity;
+  overview: string;
+  moduleName: string;
+}
+
+/**
+ * Data to build the action report summary
+ */
+export interface OtterAuditReport {
+  nbVulnerabilities: number;
+  errors: OtterAdvisory[];
+  warnings: OtterAdvisory[];
+  highestSeverityFound?: Severity;
+}
+
+export type OtterAuditReporter = (response: string, severityThreshold: Severity) => OtterAuditReport;
+
+/**
+ * Format the response from npm audit V1 in a common interface that will be used to build the
+ * report summary
+ * @param response
+ * @param severityThreshold
+ */
+export const computeNpmV1Report: OtterAuditReporter = (response: string, severityThreshold: Severity) => {
+  const reportJson = JSON.parse(response) as Audit;
+  core.debug(response);
+  const nbVulnerabilities = Object.values(reportJson.metadata.vulnerabilities as { [key: string]: number } || {}).reduce((acc, curr) => acc + curr, 0);
+  let highestSeverityFound: Severity | undefined;
+  for (let index = severities.length - 1; index >= 0; index--) {
+    const severity: Severity = severities[index];
+    if (reportJson.metadata.vulnerabilities[severity] > 0) {
+      highestSeverityFound = severity;
+      break;
+    }
+  }
+  return Object.values(reportJson.advisories as Readonly<Record<GitHubAdvisoryId, Advisory>>)
+    .reduce<OtterAuditReport>((currentVulnerabilities, advisory: Advisory) => {
+      core.info(`${severities.indexOf(severityThreshold)} - ${severities.indexOf(advisory.severity)}`);
+      if (severities.indexOf(severityThreshold) <= severities.indexOf(advisory.severity)) {
+        currentVulnerabilities.errors.push({severity: advisory.severity, overview: advisory.overview, moduleName: advisory.module_name});
+      } else {
+        currentVulnerabilities.warnings.push({severity: advisory.severity, overview: advisory.overview, moduleName: advisory.module_name});
+      }
+      return currentVulnerabilities;
+    }, {errors: [], warnings: [], nbVulnerabilities, highestSeverityFound} as OtterAuditReport);
+};
+
+/**
+ * Format the response from npm audit V2 in a common interface that will be used to build the
+ * report summary
+ * @param response
+ * @param severityThreshold
+ */
+export const computeNpmV2Report: OtterAuditReporter = (response: string, severityThreshold: Severity) => {
+  const reportJson = JSON.parse(response) as NPMAuditReportV2.Audit;
+  core.debug(response);
+  const nbVulnerabilities = severities.reduce((acc, severity) => acc + (reportJson.metadata.vulnerabilities[severity] ?? 0), 0);
+  let highestSeverityFound: Severity | undefined;
+  for (let index = severities.length - 1; index >= 0; index--) {
```

Comment:
It seems that this loop is duplicated in the 4 helpers 
we may share the same code no ?